datafusion/execution/src/cache/cache_unit.rs

```diff
@@ -94,9 +94,69 @@ impl CacheAccessor<Path, Arc<Statistics>> for DefaultFileStatisticsCache {
     }
 }
 
+/// Collected files metadata for listing files.
+/// Cache will not invalided until user call remove or clear.
+#[derive(Default)]
+pub struct DefaultListFilesCache {
+    statistics: DashMap<Path, Arc<Vec<ObjectMeta>>>,
```

Comment:
the use of `statistics` as a name here seems confusing as this doesn't hold statistics, but rather paths.

Perhaps a name like `paths` would be clearer 🤔